### PR TITLE
Add `bar` scalar function

### PR DESCRIPTION
**Pull request by niyue**
* add `bar` scalar function
  * https://clickhouse.com/docs/en/sql-reference/functions/other-functions#bar
* add `py` scalar function
* remove all warnings

### CODE REVIEW

File: bar_func/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,61 @@
+use gandiva_rust_udf_macro::udf;
+
+#[udf]
```
Comment (niyue) [inline bar_func/src/lib.rs:3]: See ClickHouse's `bar` function
https://clickhouse.com/docs/en/sql-reference/functions/other-functions#bar

File: py_func/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    fn test_py() {
+        let script = "result = arg.upper()";
+        let value = py(script, "hello");
+        assert_eq!(value, "HELLO");
+    }
+
+    #[test]
+    fn test_remove_aeiou() {
+        let script = r#"import re; result = re.sub(r'[aeiou]', '', arg)"#;
+        let value = py(script, "hello");
```
Comment (niyue) [inline py_func/src/lib.rs:44]: This `py` function allows integrating with python. It is still experimental and I don't enable it by default

File: py_func/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,47 @@
+use gandiva_rust_udf_macro::udf;
+use pyo3::prelude::*;
+use pyo3::types::PyDict;
+
+#[udf]
+pub fn py(script: &str, arg: &str) -> String {
```
Comment (ZMZ91) [inline py_func/src/lib.rs:6]: Would this lead to code injection?

Comment (niyue) [reply py_func/src/lib.rs:6]: Yes, this is possible. The `py` function is not registered by default, and we could move it into a experimental repo later.

**Pull request merged**